llvm/test/CodeGen/CHERI-Generic/Inputs/landingpad-non-preemptible.ll

```diff
@@ -0,0 +1,130 @@
+lpad:
+  %0 = landingpad { i8 addrspace(200)*, i32 }
+          catch i8 addrspace(200)* bitcast (i8 addrspace(200)* addrspace(200)* @_ZTIi to i8 addrspace(200)*)
+          catch i8 addrspace(200)* null
+  %1 = extractvalue { i8 addrspace(200)*, i32 } %0, 0
+  %2 = extractvalue { i8 addrspace(200)*, i32 } %0, 1
+  %3 = tail call i32 @llvm.eh.typeid.for(i8* addrspacecast (i8 addrspace(200)* bitcast (i8 addrspace(200)* addrspace(200)* @_ZTIi to i8 addrspace(200)*) to i8*)) #3
+  %matches = icmp eq i32 %2, %3
+  %4 = tail call i8 addrspace(200)* @__cxa_begin_catch(i8 addrspace(200)* %1) nounwind
+  br i1 %matches, label %catch1, label %catch
+
+catch1:
+  tail call void @__cxa_end_catch() nounwind
+  br label %return
+
+catch:
+  tail call void @__cxa_end_catch()
+  br label %return
+
+return:                                           ; preds = %entry, %catch1, %catch
+  %retval.0 = phi i64 [ 1, %catch1 ], [ 2, %catch ], [ %call, %entry ]
+  ret i64 %retval.0
+}
+
+declare dso_local i64 @_Z3foov() local_unnamed_addr addrspace(200)
+
+declare dso_local i32 @__gxx_personality_v0(...) addrspace(200)
+
+declare i32 @llvm.eh.typeid.for(i8*) addrspace(200) nounwind readnone
+
+declare dso_local i8 addrspace(200)* @__cxa_begin_catch(i8 addrspace(200)*) local_unnamed_addr addrspace(200)
+
+declare dso_local void @__cxa_end_catch() local_unnamed_addr addrspace(200)
+
+; UTC_ARGS: --disable
+; CHECK: .Lfunc_end0:
+; CHECK-NEXT: .size _Z8do_catchv, .Lfunc_end0-_Z8do_catchv
+; CHECK-NEXT: .local  .L_Z8do_catchv$eh_alias
+; CHECK-NEXT: .type  .L_Z8do_catchv$eh_alias,@function
+; CHECK-NEXT: .set  .L_Z8do_catchv$eh_alias, _Z8do_catchv
+; CHECK-NEXT: .size  .L_Z8do_catchv$eh_alias, .Lfunc_end0-_Z8do_catchv
+
+; CHECK:      GCC_except_table0:
+; CHECK-NEXT: .Lexception0:
+; CHECK-NEXT:  .byte 255                             # @LPStart Encoding = omit
+; CHECK-NEXT:  .byte 155                             # @TType Encoding = indirect pcrel sdata4
+; CHECK-NEXT:  .uleb128 .Lttbase0-.Lttbaseref0
+; CHECK-NEXT: .Lttbaseref0:
+; RISC-V uses DW_EH_PE_udata4 instead of uleb128 since uleb128 causes issues with linker relaxations.
+@IF-RISCV@; CHECK-NEXT:     .byte 3                   # Call site Encoding = udata4
+@IFNOT-RISCV@; CHECK-NEXT:  .byte 1                   # Call site Encoding = uleb128
+; CHECK-NEXT:  .uleb128 .Lcst_end0-.Lcst_begin0
+; CHECK-NEXT: .Lcst_begin0:
+; CHECK-NEXT:  [[CS_DIRECTIVE:(\.uleb128)|(\.word)]] .Ltmp0-.Lfunc_begin0           # >> Call Site 1 <<
+; CHECK-NEXT:  [[CS_DIRECTIVE]] .Ltmp1-.Ltmp0                  #   Call between .Ltmp0 and .Ltmp1
+; Note: RISC-V uses DW_EH_PE_udata4, so the 0xc marker uses 4 bytes instead of 1
+; CHECK-NEXT:  [[SMALL_CS_DIRECTIVE:(\.byte)|(\.word)]] 12     # (landing pad is a capability)
+; Note: the following line should not be using _Z8do_catchv, but a local alias
+; CHECK-NEXT:  .chericap  .L_Z8do_catchv$eh_alias + .Ltmp2-.Lfunc_begin0   #     jumps to .Ltmp2
```

Comment:
Bit weird to use .Lfunc_begin0 here now we have the alias but it's not _wrong_. I guess it's fine because the earlier fields of the call site use those symbols and we don't want to change things more than we have to.